### ISSUE TITLE
Issues when creating file with uninlined images

### ISSUE THREAD

**Issue by rbrito** [qpdf/qpdf#392]
Dear Jay,

I have a few files (granted, they are the minority) that present some problems when I try to uncompress them with the following command line:

```bash
qpdf --force-version=1.4 --qdf --stream-data=uncompress --compress-streams=n --decode-level=specialized --ii-min-bytes=0 --externalize-inline-images p-011.pdf p-011.qdf.pdf
```

I am using qpdf version 9.1.0 that you uploaded to Debian, with all the things that come with it, in a Debian (testing) environment.

Both poppler (and readers based on it, like gnome's PDF viewer, MATE's viewer and tools based on it, like diffpdf) and mupdf don't show the images that are there to show mathematical notation.

Here are the two files mentioned above:

[p-011.pdf](https://github.com/qpdf/qpdf/files/4033742/p-011.pdf)
[p-011.unc.pdf](https://github.com/qpdf/qpdf/files/4033743/p-011.unc.pdf)

I started stripping the command line and discovered that if I leave out the option `--ii-min-bytes=0`, then, the file is readable without any problems.

Otherwise, I get errors like the following:

```
$ diffpdf p-011.pdf p-011.unc.pdf 
"Error: Bad color space 'R190'"
"Error (10040): Bad image parameters"
"Error: Bad color space 'R190'"
"Error (10085): Bad image parameters"
"Error: Bad color space 'R190'"
"Error (10846): Bad image parameters"
"Error: Bad color space 'R190'"
"Error (13102): Bad image parameters"
"Error: Bad color space 'R190'"
"Error (13147): Bad image parameters"
"Error: Bad color space 'R190'"
"Error (10040): Bad image parameters"
"Error: Bad color space 'R190'"
"Error (10085): Bad image parameters"
"Error: Bad color space 'R190'"
"Error (10846): Bad image parameters"
"Error: Bad color space 'R190'"
"Error (13102): Bad image parameters"
"Error: Bad color space 'R190'"
"Error (13147): Bad image parameters"
"Error: Bad color space 'R190'"
"Error (10040): Bad image parameters"
"Error: Bad color space 'R190'"
"Error (10085): Bad image parameters"
"Error: Bad color space 'R190'"
"Error (10846): Bad image parameters"
"Error: Bad color space 'R190'"
"Error (13102): Bad image parameters"
"Error: Bad color space 'R190'"
"Error (13147): Bad image parameters"
```

If I use mupdf instead (which is used by a respectable number of Android Apps to render PDFs), then the error that I get on my system is:

```
$ mupdf p-011.unc.pdf 
error: invalid page object
error: invalid page object
error: invalid page object
error: invalid page object
error: invalid page object
error: unknown colorspace: R190
error: unknown colorspace: R190
error: unknown colorspace: R190
error: unknown colorspace: R190
error: unknown colorspace: R190
mupdf: warning: Errors found on page
```

And the page shows some blank spots where the images are supposed to be. I have noticed this issue with a low percentage of files (say, 15% of my collection), but these files are always there... And I frequently workaround the problem (or ignore it, throwing away the uncompressed output), but this time I saved one offending file to use here to file an issue.

Oh, pdfimages is based on poppler, but it says that the errors are syntactical (I have not yet studied manipulating images inside PDF files to know well the standard):

```
$ pdfimages -list p-011.unc.pdf 
page   num  type   width height color comp bpc  enc interp  object ID x-ppi y-ppi size ratio
--------------------------------------------------------------------------------------------
Syntax Warning: Bad color space 'R190'
Syntax Error (10040): Bad image parameters
Syntax Warning: Bad color space 'R190'
Syntax Error (10085): Bad image parameters
Syntax Warning: Bad color space 'R190'
Syntax Error (10846): Bad image parameters
Syntax Warning: Bad color space 'R190'
Syntax Error (13102): Bad image parameters
Syntax Warning: Bad color space 'R190'
Syntax Error (13147): Bad image parameters
```

Well, that's it for now. If you want further information from me, please let me know and I will try my best to provide anything that you want from me so that this bug can be killed.

Thanks,

Rogério Brito.

**Comment by jberkenbilt** [comment]
Thanks for your, as always, high quality bug report. Yes, we'll get this one properly dispatched. I'm hoping to release 9.1.1 soon. I have a few small enhancements I'd like to get out before Ubuntu 20.04 feature freeze.

**Comment by jberkenbilt** [comment]
Good news -- this has an easy fix. I will get the fix into 9.1.1.
